cylc/flow/task_job_mgr.py

```diff
@@ -483,6 +487,16 @@ def submit_task_jobs(self, workflow, itasks, curve_auth,
                     )
                 continue
 
+            if itask.killed_in_job_prep:
+                itask.waiting_on_job_prep = False
+                itask.killed_in_job_prep = False
+                itask.state_reset(TASK_STATUS_SUBMIT_FAILED)
+                self.data_store_mgr.delta_task_state(itask)
+                itask.local_job_file_path = None  # reset for retry
+                self._prep_submit_task_job_error(
+                    workflow, itask, '(killed in job prep)', '')
```

Comment:
Another problem is this does not get run for tasks that are `cylc remove`d in the preparing state, as they are removed from the pool and do not get to the submission stage